src/coreclr/classlibnative/bcltype/arraynative.cpp

```diff
@@ -115,26 +115,18 @@ FCIMPLEND
 
 
 // Returns an enum saying whether you can copy an array of srcType into destType.
-ArrayNative::AssignArrayEnum ArrayNative::CanAssignArrayType(const BASEARRAYREF pSrc, const BASEARRAYREF pDest)
+ArrayNative::AssignArrayEnum ArrayNative::CanAssignArrayType(const TypeHandle srcTH, const TypeHandle destTH)
 {
     CONTRACTL
     {
         THROWS;
         GC_TRIGGERS;
-        MODE_COOPERATIVE;
-        PRECONDITION(pSrc != NULL);
-        PRECONDITION(pDest != NULL);
+        MODE_ANY;
+        PRECONDITION(srcTH != NULL);
+        PRECONDITION(destTH != NULL);
```

Comment:
What overload is being used to convert the `TypeHandle`s into pointers? I don't think this is what we want. Perhaps negate `IsNull()` instead?

Reply:
They are retrieved from `MethodTable.ElementType` as pointers, then `TypeHandle::FromPtr` in native.

Reply:
```suggestion
        PRECONDITION(!srcTH.IsNull());
        PRECONDITION(!destTH.IsNull());
```

I see that, but the type in this signature is `TypeHandle`, which is not a pointer.

Reply:
I see what is going on. The compiler is implicitly creating a `TypeHandle` from the `NULL` and then the member `operator==` is used. This is less than idea. Using the `IsNull()` is the appropriate way to handle this check.